pulpcore/pulpcore/app/viewsets/content.py

```diff
@@ -79,3 +81,25 @@ class ArtifactViewSet(CreateDestroyReadNamedModelViewSet):
     queryset = Artifact.objects.all()
     serializer_class = ArtifactSerializer
     filter_class = ArtifactFilter
+
+    def destroy(self, request, pk):
```

Comment:
This kind of "look-and-see" logic cannot be used in concurrent applications without some kind of mutual exclusion.  It is subject to race conditions.

Reply:
True. Thanks!